config-model/src/main/java/com/yahoo/vespa/model/application/validation/UrlConfigValidator.java

```diff
@@ -0,0 +1,49 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.model.application.validation;
+
+import com.yahoo.config.model.deploy.DeployState;
+import com.yahoo.vespa.model.VespaModel;
+import com.yahoo.vespa.model.container.ApplicationContainerCluster;
+
+/**
+ * Validates that config using s3:// urls is used in public system and with nodes that are exclusive.
+ *
+ * @author hmusum
+ */
+public class UrlConfigValidator extends Validator {
+
+    @Override
+    public void validate(VespaModel model, DeployState state) {
+        if (! state.isHostedTenantApplication(model.getAdmin().getApplicationType())) return;
+
+        model.getContainerClusters().forEach((__, cluster) -> {
+            var isExclusive = hasExclusiveNodes(model, cluster);
+            validateS3UlsInConfig(state, cluster, isExclusive);
+        });
+    }
+
+    private static boolean hasExclusiveNodes(VespaModel model, ApplicationContainerCluster cluster) {
+        return model.hostSystem().getHosts()
+                .stream()
+                .flatMap(hostResource -> hostResource.spec().membership().stream())
+                .filter(membership -> membership.cluster().id().equals(cluster.id()))
+                .anyMatch(membership -> membership.cluster().isExclusive());
+    }
+
+    private static void validateS3UlsInConfig(DeployState state, ApplicationContainerCluster cluster, boolean isExclusive) {
+        if (hasUrlInConfig(cluster)) {
+            // TODO: Would be even better if we could add which config/field the url is set for in the error message
+            String message = "Found s3:// urls in config for container cluster " + cluster.getName();
```

Comment:
Now that `hasUrlInConfig()` does not check for scheme, we should probably make the error message scheme-neutral too.

Reply:
Changed this to only consider s3 urls in validator